wkhtmltopdf-image/src/main/docker/app.py

```diff
@@ -0,0 +1,123 @@
+#! /usr/bin/env python
+import json
+import os
+import tempfile
+
+from executor import execute
+from werkzeug.exceptions import HTTPException, NotFound
+from werkzeug.routing import Map, Rule
+from werkzeug.wrappers import Request, Response
+from werkzeug.wsgi import wrap_file
+
+"""
+This python server is responsible for generating PDFs from HTML
+using an underlying wkhtmltopdf command-line tool.
+
+To use this application, the user must send a POST request with
+base64 or form encoded encoded HTML content and the wkhtmltopdf Options in
+request data, with keys 'contents' and 'options'.
+The application will return a response with the PDF file.
+"""
+class WkHtmlToPdfServer():
+    def __init__(self):
+        self.url_map = Map([
+            Rule('/', endpoint='generate_pdf'),
+            Rule('/health', endpoint='health_check')
+        ])
+
+    """
+    Generate a PDF from submitted HTML.
+    """
+    def on_generate_pdf(self, request):
+        if request.method != 'POST':
+            return
+
+        request_is_json = request.content_type.endswith('json')
+
+        with tempfile.NamedTemporaryFile(suffix='.html') as source_file:
```

Comment:
I'm going to continue working on removing the file-system usage in favor of using stdin/stdout piping to/from the wkhtmltopdf CLI